design/ceph/object/zone.md

```diff
@@ -80,9 +81,12 @@ After an ceph-object-store is configured to be in this ceph-object-zone, the all
 
 At the moment creating an ceph-object-zone resource only handles Day 1 initial configuration for the zone.
 
-Changes made to the resource's configuration or deletion of the resource are not reflected on the Ceph cluster.
+By default when a CephObjectZone is deleted or modified, the poos supporting the zone are not deleted from the Ceph cluster.
 
-To be clear, when the ceph-object-zone resource is deleted or modified, the zone is not deleted from the Ceph cluster. Zone deletion must be done through the toolbox.
+If user disables `preservePoolsOnDelete` then zone or pool will be removed only if following conditions satisfied:
+- all the objectstores belong to this zone need to deleted first.
+- it should not be [master zone](/design/ceph/object/ceph-multisite-overview.md#Master-Zone/Zonegroup).
+- if it is master zone, then no other zones belongs to zonegroup.
 
```

Comment:
@cbodley: Is okay to remove pools of zone if above conditions satisy? This is mostly executed in case of clean-up. All the data residing on the pool will be removed when `objectstores` are removed. Do I need to add more conditions here?

Reply:
these look like a good set of conditions, but it's not clear to me where these are being enforced

if there are still objectstores referring to the zone resource, we should prevent the zone resource from deleted in the first place

if we're trying to delete a zone resource and find that there are other zones in its period configuration, we should remove that zone from its zonegroup and commit the period

if all of the above is enforced, then i think it's fine to delete pools during zone resource deletion

Reply:
We are checking dependency before deleting the zone in [CephObjectZoneDependents](https://github.com/rook/rook/pull/10728/files#diff-2acd79f015dc6bbe01d58404a344555269d205e7ca287996bd11595794dd196cR35) and zone deletion will be blocked until dependency are removed